### PR TITLE
Datatrails: Metric as breadcrumb and style improvements

### PR DESCRIPTION
**Pull request by adrapereira**
**What is this feature?**

This PR removes the last remnants of usage of "trail" in the UI, replacing it with the metric name in the breadcrumb and with "History" next to the history steps component.

Also adds a header to the trail page for first time users.

Lastly contains some style tweaks as proposed by @catherineymgui.

### CODE REVIEW

**Comment by catherineymgui** [comment]
Just a small nit - could we decrease the padding b/w the data source select and the search bar? (16 px -> 8)

Also in the description - could we change "Navigate through" to "Explore" as per our meeting today?

Otherwise looks good! 

Screenshot 2024-01-31 at 13 36 46

File: public/app/features/trails/MetricSelectScene.tsx [hunk 1]
```diff
@@ -298,11 +298,11 @@ export class MetricSelectScene extends SceneObjectBase<MetricSelectSceneState> {
       <div className={styles.container}>
         {showHeading && (
```
Comment (darrenjaneczek) [inline public/app/features/trails/MetricSelectScene.tsx:299]: One consequence of having the Text be part of the {showHeading} block is that it doesn't show up in the "Related Metrics" variant of this view.

image

File: public/app/features/trails/DataTrail.tsx [hunk 2]
```diff
@@ -171,9 +171,14 @@ export class DataTrail extends SceneObjectBase<DataTrailState> {
...
     const { controls, topScene, history } = model.useState();
     const styles = useStyles2(getStyles);
+    const showHeaderForFirstTimeUsers = getTrailStore().recent.length < 2;
 
     return (
       <div className={styles.container}>
+        { showHeaderForFirstTimeUsers && <Stack direction="column" gap={1}>
+          <Text variant="h1">Metrics</Text>
+          <Text color="secondary">Navigate through your Prometheus-compatible metrics without writing a query</Text>
+        </Stack>}
```
Comment (darrenjaneczek) [inline public/app/features/trails/DataTrail.tsx:181]: A common component could be shown after `showHeaderForFirstTimeUsers &&` that contains the same duplicated content from #81496

File: public/app/features/trails/MetricSelectScene.tsx (hunk 1 above)
Comment (darrenjaneczek) [inline public/app/features/trails/MetricSelectScene.tsx:299]: If we get rid of this "heading" and "show heading" we could consider using the `<Field>` wrapper component as before to simplify things.

This is assuming we want this to appear the same when it's at the main "Select Metric" mode and when it's in the "Related Metrics" tab.

**Review by darrenjaneczek — APPROVED** [review]
Minor points, quickly fixed.

I just wanted to verify with @catherineymgui:
1. For point (4) we want the "Search Metrics" field label to appear in both variants
2. How do we feel about "Explore Metrics" to be the title of the page (for point 2) instead of just "Metrics"?

**Pull request merged**